### PR TITLE
Add skip validation option to trunk push

### PR DESCRIPTION
**Pull request by dnkoutso**
Related to https://github.com/CocoaPods/CocoaPods/pull/6420

### CODE REVIEW

**Comment by segiddins** [comment]
Needs a rebase and probably deserves a spec (and a changelog entry)

**Comment by dnkoutso** [comment]
Changelog entry was there :(, updated `Gemfile.lock` to point to latest CocoaPods and pushed.

**Comment by segiddins** [comment]
Sorry, mustve missed it :(

File: spec/command/trunk/push_spec.rb [hunk 1]
```diff
@@ -170,6 +170,13 @@ def found_podspec_among_files(files)
         e = should.raise(Informative) { cmd.send(:validate_podspec) }
         e.message.should.include 'The spec did not pass validation, due to failure_reason.'
       end
+
+      it 'passes skip import validation' do
+        Validator.any_instance.expects(:skip_import_validation)
```
Comment (dnkoutso) [inline spec/command/trunk/push_spec.rb]: @segiddins what am I doing so wrong and this fails?

```
Bacon::Error: not all expectations were satisfied
unsatisfied expectations:
- expected exactly once, not yet invoked: #<AnyInstance:Pod::Validator>.skip_import_validation(any_parameters)
```

I understand the error msg and printing out Validator shows that `skip_import_validation` is correctly set.......

Comment (dnkoutso) [inline spec/command/trunk/push_spec.rb]: Code in `push.rb` is:

```ruby
def validate_podspec
          UI.puts 'Validating podspec'.yellow

          validator = Validator.new(spec, %w(https://github.com/CocoaPods/Specs.git))
          validator.allow_warnings = @allow_warnings
          validator.use_frameworks = @use_frameworks
          validator.swift_version  = @swift_version if Validator.method_defined?(:swift_version=)
          validator.skip_import_validation = @skip_import_validation
          validator.validate
          unless validator.validated?
            raise Informative, "The spec did not pass validation, due to #{validator.failure_reason}."
          end

          # Let the validator's logic for the swift version
          # set the value for the trunk JSON uploader
          @swift_version = validator.respond_to?(:used_swift_version) && validator.used_swift_version
        end
```

Comment (segiddins) [inline spec/command/trunk/push_spec.rb]: `expects(:skip_import_validation=)` ?

Comment (segiddins) [inline spec/command/trunk/push_spec.rb]: And you probably want a `.with(true)` as well

Comment (dnkoutso) [inline spec/command/trunk/push_spec.rb]: :man_facepalming:

Comment (dnkoutso) [inline spec/command/trunk/push_spec.rb]: 🙏

**Comment by dnkoutso** [comment]
All set.

**Pull request merged**